File: go/array/48_rotate_.go
package array

// rotate 旋转图像
func rotate2(matrix [][]int) {

	// 水平翻转 + 对角线翻转  == 顺时针旋转90°

	n := len(matrix)

	for i := 0; i < n/2; i++ {
		matrix[i], matrix[n-i-1] = matrix[n-i-1], matrix[i]
	}

	for i := 0; i < n; i++ {
		for j := 0; j < i; j++ {
			matrix[i][j], matrix[j][i] = matrix[j][i], matrix[i][j]
		}
	}

}

File: rust/src/array/48_rotate.rs



// 旋转图像(二维数组)
pub fn rotate(matrix: &mut Vec<Vec<i32>>) {

    // 水平翻转 + 对角线翻转
    let n = matrix.len();

    for i in 0..n/2 {
        // (&mut matrix[i], &mut matrix[n-i-1]) = (&mut matrix[n-i-1], &mut matrix[i]);
        matrix.swap(i, n-i-1);
    }

    for i in 0..n {
        for j in 0..i {
            (matrix[i][j], matrix[j][i]) = (matrix[j][i], matrix[i][j]);
            
        }
    }

}